srcs/parser/parser.c

```diff
@@ -3,11 +3,77 @@
 /*                                                        :::      ::::::::   */
 /*   parser.c                                           :+:      :+:    :+:   */
 /*                                                    +:+ +:+         +:+     */
-/*   By: tsuchen <tsuchen@student.42.fr>            +#+  +:+       +#+        */
+/*   By: jteissie <jteissie@student.42.fr>          +#+  +:+       +#+        */
 /*                                                +#+#+#+#+#+   +#+           */
 /*   Created: 2024/08/15 12:12:00 by tsuchen           #+#    #+#             */
-/*   Updated: 2024/08/15 14:25:48 by jteissie         ###   ########.fr       */
+/*   Updated: 2024/08/16 19:08:45 by jteissie         ###   ########.fr       */
 /*                                                                            */
 /* ************************************************************************** */
 
 #include "cub3d.h"
+
+t_parse_status	check_walls(char **map, uint32_t x, uint32_t y, size_t bound)
+{
+	if (x == bound || y >= ft_strlen(map[x]))
+		return (MAP_ERR);
+	else if (map[x][y] == 'F')
+		return (MAP_OK);
+	else if (map[x][y] == '1')
+		return (MAP_OK);
+	else if (x == 0 || map[x] == NULL)
+		return (MAP_ERR);
+	else if (y == 0 || map[x][y] == '\n' || map[x][y] == '\0')
+		return (MAP_ERR);
+	map[x][y] = 'F';
+	if (check_walls(map, x - 1, y, bound) == MAP_ERR)
+		return (MAP_ERR);
+	if (check_walls(map, x + 1, y, bound) == MAP_ERR)
+		return (MAP_ERR);
+	if (check_walls(map, x, y - 1, bound) == MAP_ERR)
+		return (MAP_ERR);
+	if (check_walls(map, x, y + 1, bound) == MAP_ERR)
+		return (MAP_ERR);
+	return (MAP_OK);
+}
+
+static void	get_player_dir(t_data *data, u_int32_t x, u_int32_t y)
+{
+	if (data->map[x][y] == 'N')
+		data->p_dir = NORTH;
+	else if (data->map[x][y] == 'S')
+		data->p_dir = SOUTH;
+	else if (data->map[x][y] == 'W')
+		data->p_dir = WEST;
+	else if (data->map[x][y] == 'E')
+		data->p_dir = EAST;
+}
+
+t_parse_status	verify_map(char **map, t_data *data)
+{
+	uint32_t	start[2];
+
+	start[0] = 0;
+	start[1] = 0;
+	if (check_invalid_chars(map) == MAP_ERR)
+		return (MAP_ERR);
+	get_player_dir(data, start[0], start[1]);
```

Comment:
Need to swap get_player_dir() after find_start() call since that is where the start position of the player is checked and validated.